core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala

```diff
@@ -156,33 +207,54 @@ object H2OSchemaUtils {
     StructType(mergedSchema.map(_.field))
   }
 
-  private def rowsToRowSchemas(df: DataFrame): Dataset[Seq[FieldWithOrder]] = {
-    implicit val encoder = org.apache.spark.sql.Encoders.kryo[Seq[FieldWithOrder]]
+  def rowsToRowSchemas(df: DataFrame): Dataset[ArrayBuffer[FieldWithOrder]] = {
+    implicit val encoder = org.apache.spark.sql.Encoders.kryo[ArrayBuffer[FieldWithOrder]]
     val originalSchema = df.schema
-    df.map[Seq[FieldWithOrder]] { row: Row =>
-      originalSchema.fields.zipWithIndex.foldLeft(Seq.empty[FieldWithOrder]) {
-        case (acc, (field, index)) => acc ++ flattenField(field, row(index), index :: Nil)
+    df.map[ArrayBuffer[FieldWithOrder]] { row: Row =>
+      var i = 0
```

Comment:
Any specific reason for using `i` instead if `zipWithIndex` and iterating together with that index?

Reply:
This is just a bit faster

Reply:
But if prefer more functional style, I can change it.

Reply:
It's fine, just wanted to know the reason :)